packages/kbn-es/src/install/archive.js

```diff
@@ -90,6 +90,16 @@ async function appendToConfig(installPath, key, value) {
  * @param {ToolingLog} log
  */
 async function configureKeystore(installPath, password, log = defaultLog) {
+  const configPath =
+    process.env.ES_PATH_CONF || path.join(installPath, 'config');
+  const keystorePath = path.join(configPath, 'elasticsearch.keystore');
+
+  if (fs.existsSync(keystorePath)) {
+    log.info('keystore already exists at %s', chalk.bold(keystorePath));
+    log.warning('unable to bootstrap password');
+    return;
+  }
+
   log.info('setting bootstrap password to %s', chalk.bold(password));
 
   await execa(ES_KEYSTORE_BIN, ['create'], { cwd: installPath });
```

Comment:
Is there a way we could instead detect from the result of `bin/elasticsearch-keystore create` that a keystone already exists? Feels like basing this all off of the `ES_PATH_CONF` environment variable seems... unexpected.

Reply:
++ for this.

Reply:
Pushed https://github.com/elastic/kibana/pull/19332/commits/be346ded3075ae52ce875ae29b69f13cbf7306a7 to account for this.